pkg/server/commands/listusers/list_users_rpc.go

```diff
@@ -117,8 +117,18 @@ func doesHavePossibleEdges(typesys *typesystem.TypeSystem, req *openfgav1.ListUs
 	g := graph.New(typesys)
 
 	userFilters := req.GetUserFilters()
-	source := typesystem.DirectRelationReference(userFilters[0].GetType(), userFilters[0].GetRelation())
-	target := typesystem.DirectRelationReference(req.GetObject().GetType(), req.GetRelation())
+
+	userFilterType := userFilters[0].GetType()
+	userFilterRelation := userFilters[0].GetRelation()
+	targetType := req.GetObject().GetType()
+	targetRelation := req.GetRelation()
+
+	if userFilterType == targetType && userFilterRelation == targetRelation {
+		return true, nil // reflexive userset
+	}
```

Comment:
This is the core functional change of this PR.

Reply:
(**suggestion**) Can we have this extracted into a func so that it better describes what happens here and is contained? Perhaps something like:

```go
if isAReflexiveUserset() {
  return true, nil
}
```

Reply:
A named function rename for a boolean check is superfluous.  I'd just ask you improve the doc comment by adding a sample tuple with the corresponding filter (e.g. `// document:1#viewer, user_filter=[{...}]`)

Reply:
Nice resolution @willvedd . I see you introduced the variable for clarity but avoided the extra function. Nice 👍 Best of both worlds.